## Figures for **Genome-wide barcoded transposon screens for cancer drug sensitivity in haploid mouse embryonic stem cells.**

```{r setup, echo=FALSE}
# To compile this document in R:
# install.packages('knitr')
# library(knitr)
# knit2html("figures.Rmd")

knitr::opts_chunk$set(dev="pdf", dpi=300)
```


### Figure 2A
```{r figure2a}
dmsos <- lapply(liblist, dmso_reads)
# Calculate pairwise correlations
rs <- unlist(lapply(dmsos, function(x) cor(x[,-1])))
# Exclude self-self
mean(rs[rs != 1])

# Example plot
plot(log(dmsos[["L2spk"]][,3],10),
  log(dmsos[["L2spk"]][,4],10),
  pch=20,
  xlab="DMSO rep 1 reads (log10)",
  ylab="DMSO rep 2 reads (log10)",
  cex.axis=2
)

```

### Figure 2C
```{r figure2c}
volcHighlight(spike_edgeanlist[["L2spksims"]],"olaparib","ACAAGATTAAAGTAAAAATAACAGA")
```

### Figure 2D
```{r figure2d}
# Code to call hits for BMN 673 sensitivity across all libraries:
bmn_hits <- dhitgenes(sens,maplist,"BMN.673",mincount=4)
# To get Ewsr1 barcodes for mutants that score as hits:
hitmappings <- lapply(names(liblist),maphits,el=edgelist,ml=maplist,drug="BMN.673")
hitmappings <- lapply(hitmappings, function(x) x[,grep("L|barcode|pos|gene|chr|log|PValue",names(x))])
ewsr1_bmn <- lapply(hitmappings, function(x) x[grep("Ewsr1",x[,"gene"]),])

ewsr1_l2_bc <- c("AAGTTCACGGTTTGCGACTTAGATG","GTCTTGAGCAAGTATGATTGCTTAG")
volcHighlight(edgeanlist[["L2spk"]],"BMN.673",ewsr1_l2_bc)
```
